### PR TITLE
feat(federation): x-com sendcoins 23: invoke settlement of pending tx

### PR DESCRIPTION
**Pull request by clauspeterhuebner**
### Issues

- None

### Todo

- [X] None

### CODE REVIEW

**Review by einhornimmond — DISMISSED** [review]

File: backend/src/graphql/resolver/util/processXComSendCoins.ts [hunk 1]
```diff
@@ -114,3 +114,95 @@ export async function processXComPendingSendCoins(
...
+        args.memo = pendingTx.memo
+        args.senderCommunityUuid = pendingTx.userCommunityUuid
+        args.senderUserUuid = pendingTx.userGradidoID
+        if (pendingTx.userName) {
+          args.senderUserName = pendingTx.userName
+        }
+        logger.debug(`X-Com: ready for settleSendCoins with args=`, args)
+        const acknoleged = await client.settleSendCoins(args)
+        logger.debug(`X-Com: returnd from settleSendCoins:`, acknoleged)
+        if (acknoleged) {
```
Comment (einhornimmond) [inline backend/src/graphql/resolver/util/processXComSendCoins.ts:177]:
```suggestion
        if (acknowledge) {
```

File: database/entity/0072-add_communityuuid_to_transactions_table/Transaction.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+import { DecimalTransformer } from '../../src/typeorm/DecimalTransformer'
+import { Contribution } from '../Contribution'
+import { DltTransaction } from '../DltTransaction'
+
+@Entity('transactions')
+export class Transaction extends BaseEntity {
+  @PrimaryGeneratedColumn('increment', { unsigned: true })
+  id: number
+
+  @Column({ type: 'int', unsigned: true, unique: true, nullable: true, default: null })
```
Comment (einhornimmond) [inline database/entity/0072-add_communityuuid_to_transactions_table/Transaction.ts:13]: Attention: 
nullable = true + unique = true:  means that only one entry with value NULL allowed to exist in db

File: database/entity/0072-add_communityuuid_to_transactions_table/Transaction.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  previous: number | null
+
+  @Column({ name: 'type_id', unsigned: true, nullable: false })
```
Comment (einhornimmond) [inline database/entity/0072-add_communityuuid_to_transactions_table/Transaction.ts:16]: nullable: false is default can also be omitted.

File: federation/src/graphql/util/checkTradingLevel.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import CONFIG from '@/config'
+import { Community as DbCommunity } from '@entity/Community'
+import { Decimal } from 'decimal.js-light'
+import { federationLogger as logger } from '@/server/logger'
+
+export async function checkTradingLevel(homeCom: DbCommunity, amount: Decimal): Promise<boolean> {
+  const tradingLevel = CONFIG.FEDERATION_TRADING_LEVEL
+  if (homeCom.url !== tradingLevel.RECEIVER_COMMUNITY_URL) {
+    logger.warn(
+      `X-Com: tradingLevel allows to receive coins only wiht url ${tradingLevel.RECEIVER_COMMUNITY_URL}`,
```
Comment (einhornimmond) [inline federation/src/graphql/util/checkTradingLevel.ts:10]:
```suggestion
      `X-Com: tradingLevel allows to receive coins only with url ${tradingLevel.RECEIVER_COMMUNITY_URL}`,
```

**Review by einhornimmond — DISMISSED** [review]

File: database/entity/0072-add_communityuuid_to_transactions_table/Transaction.ts (hunk 2 above)
Comment (clauspeterhuebner) [reply database/entity/0072-add_communityuuid_to_transactions_table/Transaction.ts:13]: seemingly not, beause this column definition still exists for a longer time.
In production database the table `transactions` contains for the column `previous` several entries with value NULL.
The definition `unique: true` exists only on entity-level, in database there should exist an unique-key-contraint and the constraints have to be actived to reach your described effect.

I don't know if on entity-level the `unique:true` will have any effect, without dedicated database support...

File: database/entity/0072-add_communityuuid_to_transactions_table/Transaction.ts (hunk 3 above)
Comment (clauspeterhuebner) [reply database/entity/0072-add_communityuuid_to_transactions_table/Transaction.ts:16]: that's right...

**Review by einhornimmond — APPROVED** [review]
I see some potential for refactoring, but looks okay so far.

**Pull request merged**